manifest.js

```diff
@@ -0,0 +1,66 @@
+const pathToRegExp = require('path-to-regexp');
+
+module.exports = {
+  version: '1.0.0',
+  policies: ['rewrite'],
+  init: function (pluginContext) {
+    pluginContext.registerPolicy({
+      name: 'rewrite',
+      policy: (actionParams) => {
+        const compiledExp = pathToRegExp.compile(actionParams.rewrite);
+
+        return (req, res, next) => {
+          let toUrl = null;
+
+          if (req.egContext.matchedCondition.plainRegEx)
+            toUrl = req.url.replace(req.egContext.matchedCondition.plainRegEx, actionParams.rewrite);
+          else
+            toUrl = compiledExp(req.egContext.matchedCondition);
+
+          if (!actionParams.redirect) {
+            req.url = toUrl;
+            return next();
+          }
+
+          res.statusCode = actionParams.redirect === 'permanent' ? 301 : 302;
+          res.setHeader('Location', toUrl);
+          res.end();
```

Comment:
why not use res.redirect(301,toUrl); ?

Reply:
Good point, I think it's handling more cases than I could imagine. I'll fix it.